private/protocol/restxml/build_bench_test.go

```diff
@@ -3,47 +3,115 @@
-		HTTPMethod: "POST",
-		HTTPPath:   "/2015-04-17/distribution/{DistributionId}/invalidation",
-	}
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials:      credentials.NewStaticCredentials("Key", "Secret", "Token"),
+		Endpoint:         aws.String(server.URL),
+		S3ForcePathStyle: aws.Bool(true),
+		DisableSSL:       aws.Bool(true),
+		Region:           aws.String(endpoints.UsWest2RegionID),
+	}))
+	cloudfrontSvc = cloudfront.New(sess)
+	s3Svc = s3.New(sess)
 
-	benchRESTXMLBuild(b, op, params)
+	c := m.Run()
+	server.Close()
+	os.Exit(c)
 }
 
-func BenchmarkRESTXMLBuild_Simple_cloudfrontDeleteStreamingDistribution(b *testing.B) {
-	params := &cloudfront.DeleteDistributionInput{
-		Id:      aws.String("string"), // Required
-		IfMatch: aws.String("string"),
-	}
-	op := &request.Operation{
-		Name:       "DeleteStreamingDistribution",
-		HTTPMethod: "DELETE",
-		HTTPPath:   "/2015-04-17/streaming-distribution/{Id}",
-	}
-	benchRESTXMLBuild(b, op, params)
+func BenchmarkRESTXMLBuild_Complex_CFCreateDistro(b *testing.B) {
+	params := cloudfrontCreateDistributionInput()
+
+	benchRESTXMLBuild(b, func() *request.Request {
+		req, _ := cloudfrontSvc.CreateDistributionRequest(params)
+		return req
+	})
 }
 
-func BenchmarkEncodingXMLMarshal_Simple_cloudfrontDeleteStreamingDistribution(b *testing.B) {
-	params := &cloudfront.DeleteDistributionInput{
-		Id:      aws.String("string"), // Required
-		IfMatch: aws.String("string"),
-	}
+func BenchmarkRESTXMLRequest_Complex_CFCreateDistro(b *testing.B) {
+	benchRESTXMLRequest(b, func() *request.Request {
+		req, _ := cloudfrontSvc.CreateDistributionRequest(cloudfrontCreateDistributionInput())
+		return req
+	})
+}
+
+func BenchmarkRESTXMLBuild_Simple_CFDeleteDistro(b *testing.B) {
+	params := cloudfrontDeleteDistributionInput()
+
+	benchRESTXMLBuild(b, func() *request.Request {
+		req, _ := cloudfrontSvc.DeleteDistributionRequest(params)
+		return req
+	})
+}
+
+func BenchmarkRESTXMLRequest_Simple_CFDeleteDistro(b *testing.B) {
+	benchRESTXMLRequest(b, func() *request.Request {
+		req, _ := cloudfrontSvc.DeleteDistributionRequest(cloudfrontDeleteDistributionInput())
+		return req
+	})
+}
+
+func BenchmarkRESTXMLBuild_REST_S3HeadObject(b *testing.B) {
+	params := s3HeadObjectInput()
+
+	benchRESTXMLBuild(b, func() *request.Request {
+		req, _ := s3Svc.HeadObjectRequest(params)
+		return req
+	})
+}
+
+func BenchmarkRESTXMLRequest_REST_S3HeadObject(b *testing.B) {
+	benchRESTXMLRequest(b, func() *request.Request {
+		req, _ := s3Svc.HeadObjectRequest(s3HeadObjectInput())
+		return req
+	})
+}
+
+func BenchmarkRESTXMLBuild_XML_S3PutObjectAcl(b *testing.B) {
+	params := s3PutObjectAclInput()
+
+	benchRESTXMLBuild(b, func() *request.Request {
+		req, _ := s3Svc.PutObjectAclRequest(params)
+		return req
+	})
+}
+
+func BenchmarkRESTXMLRequest_XML_S3PutObjectAcl(b *testing.B) {
+	benchRESTXMLRequest(b, func() *request.Request {
+		req, _ := s3Svc.PutObjectAclRequest(s3PutObjectAclInput())
```

Comment:
This was so that the `cost` of creating an API  input parameter was included in the benchmark results. The `Build` benchmarks try to narrow the focus just to building overhead.